### PR TITLE
Document project archiving

### PR DESCRIPTION
**Pull request by cpeel**
After reviewing remaining UTF-8 bits, I realized I had overlooked converting the `page_events` and `wordcheck_events` tables in the archive database. Which then lead me to realize that we don't talk about creating those tables anywhere.

This MR includes:
* A commit to convert the non-project tables in the archive database to UTF-8, since we will be inserting new values in that from UTF-8 tables in the main database.
* A new `ARCHIVING.md` file describing how to set up project archiving.

### CODE REVIEW

File: SETUP/ARCHIVING.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+3. Create the database and grant it access using the **same credentials used for
+   the primary database**. From within a `mysql` client session:
+   ```sql
+   CREATE DATABASE dp_archive CHARACTER SET utf8mb4;
+   GRANT ALL  ON dp_archive.* TO dp_user@localhost IDENTIFIED BY 'dp_password';
+   ```
+
+4. Create the `page_events` and `wordcheck_events` table in the archive
+   database. Find the schema for those tables in `db_schema.sql` and use them
```
Comment (chrismiceli) [inline SETUP/ARCHIVING.md:34]: Should we create a separate schema for these shared tables and have the db_schema.swl import it?

File: SETUP/ARCHIVING.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+   location for this is in the document root at the same level as the `projects`
+   directory. Ensure that this directory is owned and writable by the web server.
+   For example:
+   ```bash
+   mkdir /var/www/htdocs/project.archives
+   chown -R www-data:www:data /var/www/htdocs/project.archives
+   ```
+
+Now that the archive database and filesystem directory are set up, you can
+enable periodic archiving by calling `crontab/archive_projects.php` via cron.
```
Comment (chrismiceli) [inline SETUP/ARCHIVING.md:47]: I'm not a markdown expert but can we relatively link to these referenced files?

File: SETUP/ARCHIVING.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+3. Create the database and grant it access using the **same credentials used for
+   the primary database**. From within a `mysql` client session:
+   ```sql
+   CREATE DATABASE dp_archive CHARACTER SET utf8mb4;
+   GRANT ALL  ON dp_archive.* TO dp_user@localhost IDENTIFIED BY 'dp_password';
+   ```
+
+4. Create the `page_events` and `wordcheck_events` table in the archive
+   database. Find the schema for those tables in `db_schema.sql` and use them
+   to create the tables in `dp_archive`.
```
Comment (jmdyck) [inline SETUP/ARCHIVING.md:35]:
```suggestion
   database.
   Find the 'CREATE TABLE' commands for those 2 tables in `db_schema.sql`.
   These commands create those tables in the main database;
   the corresponding tables in the archive database have exactly the same structure.
   So to create them, make the archive db the current database (e.g. `use dp_archive`), and then run those 2 'CREATE TABLE' commands.
```

File: SETUP/ARCHIVING.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+5. Create the archive directory referenced in `_ARCHIVE_PROJECTS_DIR`, a common
```
Comment (jmdyck) [inline SETUP/ARCHIVING.md:37]:
```suggestion
5. Create the archive directory referenced in `_ARCHIVE_PROJECTS_DIR`. A common
```

File: SETUP/ARCHIVING.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+   location for this is in the document root at the same level as the `projects`
+   directory. Ensure that this directory is owned and writable by the web server.
```
Comment (jmdyck) [inline SETUP/ARCHIVING.md:39]:
```suggestion
   directory.
   (It doesn't have to be in the document root;
   the DP code doesn't assume that it's web-accessible.
   But you might choose to make it so [for some reason].)
   Ensure that this directory is owned and writable by the web server.
```

File: SETUP/ARCHIVING.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+   ```bash
+   mkdir /var/www/htdocs/project.archives
+   chown -R www-data:www:data /var/www/htdocs/project.archives
+   ```
+
+Now that the archive database and filesystem directory are set up, you can
+enable periodic archiving by calling `crontab/archive_projects.php` via cron.
+The `dp.cron` crontab file includes a commented-out example.
+
+`crontab/archive_projects.php` controls which projects are archived, and the
```
Comment (jmdyck) [inline SETUP/ARCHIVING.md:50]:
```suggestion
`crontab/archive_projects.php` controls which projects are archived (and when), and the
```

File: SETUP/ARCHIVING.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+logic to do the archiving is `pinc/archiving.inc`.
```
Comment (jmdyck) [inline SETUP/ARCHIVING.md:51]:
```suggestion
logic to do the archiving is in `pinc/archiving.inc`.
```

File: SETUP/ARCHIVING.md (hunk 1 above)
Comment (cpeel) [reply SETUP/ARCHIVING.md:34]: Unfortunately we have several tools that use the `db_schema.sql` directly, so this wouldn't work.

File: SETUP/ARCHIVING.md (hunk 2 above)
Comment (cpeel) [reply SETUP/ARCHIVING.md:47]: Yup, sure can.

File: SETUP/ARCHIVING.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+     this document we use the name `dp_archive` for this value.
+
+   * `_ARCHIVE_PROJECTS_DIR` sets the directory where project filesystem
+     artifacts are moved.
+
+2. Run the `configure` script to update the site configuration with the new
+   archive settings. See [INSTALL.md](INSTALL.md) for more information on how
+   to do this.
+
+3. Create the database and grant it access using the **same credentials used for
```
[the remark this answers is not in the document]
Comment (cpeel) [reply SETUP/ARCHIVING.md:26]: 👍

File: SETUP/ARCHIVING.md [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+   the primary database**. From within a `mysql` client session:
```
[the remark this answers is not in the document]
Comment (cpeel) [reply SETUP/ARCHIVING.md:27]: Good call.

File: SETUP/ARCHIVING.md (hunk 3 above)
Comment (cpeel) [reply SETUP/ARCHIVING.md:35]: Much clearer - thanks!

File: SETUP/ARCHIVING.md (hunk 4 above)
Comment (cpeel) [reply SETUP/ARCHIVING.md:37]: Done.

File: SETUP/ARCHIVING.md (hunk 5 above)
Comment (cpeel) [reply SETUP/ARCHIVING.md:39]: 👍

File: SETUP/ARCHIVING.md (hunk 7 above)
Comment (cpeel) [reply SETUP/ARCHIVING.md:51]: Added.

File: SETUP/ARCHIVING.md (hunk 6 above)
Comment (cpeel) [reply SETUP/ARCHIVING.md:50]: Good call.

File: SETUP/upgrade/14/20200511_convert_archive_db_to_utf8mb4.php [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+
+$result = mysqli_query(DPDatabase::get_connection(), $sql) or die( mysqli_error(DPDatabase::get_connection()) );
+
+$latin1_tables = array();
+while($row = mysqli_fetch_assoc($result))
+{
+    $latin1_tables[] = $row['table_name'];
+}
+
+$stop_file = '/tmp/utf8_conversion';
```
[the remark this answers is not in the document]
Comment (cpeel) [reply SETUP/upgrade/14/20200511_convert_archive_db_to_utf8mb4.php:73]: This is the same stop-file name we use in the script that does the project table conversion, so I'd like to keep them the same. And if they create such a file the script will just stop before it starts and they can rename their note-taking script 😁

**Comment by cpeel** [comment]
> Generally looks good. I just made some suggestions.

Thank you for the excellent, detailed feedback @jmdyck - much appreciated.

File: SETUP/upgrade/14/20200511_convert_archive_db_to_utf8mb4.php (hunk 10 above)
[the remark this answers is not in the document]
Comment (jmdyck) [reply SETUP/upgrade/14/20200511_convert_archive_db_to_utf8mb4.php:73]: > This is the same stop-file name we use in the script that does the project table conversion, so I'd like to keep them the same.

Okay.

> And if they create such a file the script will just stop before it starts and they can rename their note-taking script grin

I was more thinking of the case where someone happens to create the file after the converter script has started. To make it worse, imagine that the person who started the converter has stepped away (for a meal or some sleep), so doesn't notice that it has now stopped. So the upgrade process could lose a few hours. The whole scenario is fairly unlikely, but on the other hand the 'fix' was tiny, so I thought it was worth suggesting.

**Pull request merged**